mp/src/game/client/neo/ui/neo_scoreboard.cpp

```diff
@@ -1,255 +1,759 @@
+	wide = GetWide();
+	m_pJinraiPlayerList->SetSize(wide / 2, tallest);
+	m_pNSFPlayerList->SetSize(wide / 2, tallest);
+	if (m_iDesiredHeight < tall)
 	{
-		m_pPlayerList->SetSelectedItem(selectedRow);
+		SetSize(wide, tall);
+		m_pSpectatorsPlayerList->SetSize(wide, spectatorsTall);
 	}
-
-	// Update team headers
-	auto *teamJinrai = GetGlobalTeam(TEAM_JINRAI);
-	auto *teamNSF = GetGlobalTeam(TEAM_NSF);
-	if (teamJinrai && teamNSF)
+	else
 	{
-		char szTeamHeaderText[256];
-		wchar_t wszTeamHeaderText[256];
+		SetSize(wide, m_iDesiredHeight);
+		m_pSpectatorsPlayerList->SetSize(wide, m_iDesiredHeight - tallest);
+	}
 
-		memset(szTeamHeaderText, 0, sizeof(szTeamHeaderText));
-		V_snprintf(szTeamHeaderText, sizeof(szTeamHeaderText), "JINRAI       Score: %d    Players: %d", teamJinrai->GetRoundsWon(), teamCountJinrai);
-		g_pVGuiLocalize->ConvertANSIToUnicode(szTeamHeaderText, wszTeamHeaderText, sizeof(wszTeamHeaderText));
-		m_pPlayerList->ModifyColumn(SCORESECTION_JINRAI, "name", wszTeamHeaderText);
+	m_pSpectatorsPlayerList->SetPos(0, tallest + addHeight);
 
-		memset(szTeamHeaderText, 0, sizeof(szTeamHeaderText));
-		V_snprintf(szTeamHeaderText, sizeof(szTeamHeaderText), "NSF          Score: %d    Players: %d", teamNSF->GetRoundsWon(), teamCountNSF);
-		g_pVGuiLocalize->ConvertANSIToUnicode(szTeamHeaderText, wszTeamHeaderText, sizeof(wszTeamHeaderText));
-		m_pPlayerList->ModifyColumn(SCORESECTION_NSF, "name", wszTeamHeaderText);
-	}
+	MoveToCenterOfScreen();
 
-	m_pPlayerList->SetVisible(true);
+	// update every second
+	m_fNextUpdateTime = gpGlobals->curtime + 1.0f;
 }
 
+//-----------------------------------------------------------------------------
+// Purpose: Sort all the teams
+//-----------------------------------------------------------------------------
 void CNEOScoreBoard::UpdateTeamInfo()
 {
-	BaseClass::UpdateTeamInfo();
-}
+	if ( !g_PR )
+		return;
 
-bool CNEOScoreBoard::GetPlayerScoreInfo(int playerIndex, KeyValues *outPlayerInfo)
-{
-	return BaseClass::GetPlayerScoreInfo(playerIndex, outPlayerInfo);
+	int iNumPlayersInGame = 0;
+
+	for ( int j = 1; j <= gpGlobals->maxClients; j++ )
+	{
+		if ( g_PR->IsConnected( j ) )
+		{
+			iNumPlayersInGame++;
+		}
+	}
+
+	// update the team sections in the scoreboard
+	for ( int i = TEAM_SPECTATOR; i < TEAM_MAXCOUNT; i++ )
+	{
+		wchar_t *teamName = NULL;
+		int sectionID = 0;
+		C_Team *team = GetGlobalTeam(i);
+
+		if ( team )
+		{
+			auto pPlayerList = GetPanelForTeam( i );
+			sectionID = GetSectionFromTeamNumber( i );
+
+			// update team name
+			wchar_t name[64];
+			wchar_t string1[1024];
+			wchar_t wNumPlayers[6];
+
+			if ( HL2MPRules()->IsTeamplay() == false )
+			{
+				_snwprintf( wNumPlayers, ARRAYSIZE(wNumPlayers), L"%i", iNumPlayersInGame );
+#ifdef WIN32
+				_snwprintf( name, ARRAYSIZE(name), L"%s", g_pVGuiLocalize->Find("#ScoreBoard_Deathmatch") );
+#else
+				_snwprintf( name, ARRAYSIZE(name), L"%S", g_pVGuiLocalize->Find("#ScoreBoard_Deathmatch") );
+#endif
```

Comment:
Is there some reason we need to specify `%s` for Windows here instead of `%S` (or `%ls`)? Valve's print format API also seems to provide a `V_snwprintf` which evaluates to something like
```cpp
#ifdef _WIN32
	int len = _vsnwprintf( pDest, maxLen, pFormat, params );
#elif POSIX
	int len = vswprintf( pDest, maxLen, pFormat, params );
```

Reply:
To be honest I have zero experience with wide characters in C. I just did what the Valve programmer did in the if-case above. But if %S works on Windows we should probably use that with V_snwprintf.

Reply:
The Linux manpage for snprintf also comments tersely:

> **S**
> 
> (Not in C99, but in SUSv2.) Synonym for ls. Don't use. 

in favor of `%ls` I suppose. But I don't know what their rationale for this remark is.

and:

> **l**
> 
> (ell) A following integer conversion corresponds to a long int or unsigned long int argument, or a following n conversion corresponds to a pointer to a long int argument, or a following c conversion corresponds to a wint_t argument, **or a following s conversion corresponds to a pointer to wchar_t argument**. 

So that's also an option. I assume there must've been some incompatibility with the Win API at some point, leading to those ifdefs, but whether that's still an issue, no idea.

Reply:
%ls seems to work fine on Windows, could you try it on Linux with the latest commit?
I'm not able to reproduce the cut off "Deaths" issue. For me it looks like my screenshots in the OP, so not sure what's going on there 🤔

Reply:
> I'm not able to reproduce the cut off "Deaths" issue. For me it looks like my screenshots in the OP, so not sure what's going on there 🤔

@AdamTadeusz If you're building for Windows platform, would you mind testing if you can reproduce the "Deaths" scoreboard label cutting off at 1280x720 resolution, as seen in: https://github.com/NeotokyoRebuild/neo/pull/347#pullrequestreview-2108905197 for this PR?